src/events/modalSubmitInteractionCreateEventHandler.ts

```diff
@@ -0,0 +1,47 @@
+import { Collection, Interaction } from 'discord.js';
+import { EventHandler } from './eventHandler';
+import Client from '#src/Client';
+import { logger } from '#util/logger';
+import ModalSubmitInteractionHandlers from '../interactions/modal';
+import { ModalSubmitInteractionHandler } from '../interactions/modal/modalInteractionHandler';
+
+export class modalSubmitInteractionCreateEventHandler implements EventHandler {
+    readonly eventName = 'interactionCreate';
+    readonly client: Client;
+    readonly modalSubmitInteractionHandlers = new Collection<string, ModalSubmitInteractionHandler>();
+
+    constructor(client: Client) {
+        this.client = client;
+    }
+
+    async execute(interaction: Interaction): Promise<void> {
+        if (!interaction.isModalSubmit()) return;
+
+        const client = this.client;
+
+        for (const InteractionHandler of ModalSubmitInteractionHandlers) {
+            const interactionHandler = new InteractionHandler(client);
+            this.modalSubmitInteractionHandlers.set(interactionHandler.customId, interactionHandler);
+        }
```

Comment:
This seems like something that should be moved to the constructor, as we definitely don't need to re-build this every time we receive a modal submit interaction.